### PR TITLE
Issue 5: Join and share lists

### LINKED ISSUE #5: 5. As a user, I want to join an existing shopping list so I can share a shopping list with another person
## Summary
We want to allow users to join existing shopping lists so they can manage them with friends or family. This would take the following steps:
1. User enters an existing token into a form
2. The token is saved to `localStorage`
3. The user is taken to the `List` view and shown the existing list.

## Acceptance criteria
If a user **doesn’t** already have a token:
- [x] The `Home` view shows a form that allows the user to enter a token to join an _existing_ list, in addition to the button that allows them to create a _new_ list.
- [x] The input that accepts the list token has a semantic `label` element associated with its
- [x] The user can submit this form with both the mouse and the `Enter` key
- [x] If the list exists,
	- the token is saved in `localStorage`
	- the user is redirected to the `List` view and shown the items on that list
- [x] If the list _does not_ exist, the user is shown an error message that explains the problem

If a user **does** already have a token:
- [x] They are automatically redirected to the `List` view.

## Notes
- Be sure to use the `setListToken` function to save the token that the user enters!
- You will know your `<label>` is correctly implemented if you click on the label and keyboard focus moves to the related input. Refer to [the MDN docs on `<label>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label) for more info.
- Users will be able to submit the form with the `Enter` key if you listen for the right JavaScript event on the right element! If you're adding an `onKeyDown` listener anywhere, you might be overcomplicating things.
- You can assume that users will only be sharing a token if the list has actual items on it. So, let's say I create a New Shopping List, but never add any items to it, and haven't saved that particular token to Firestore, we can assume that's a list that doesn't exist.
- If you would like, you can save the token to Firestore immediately when creating a list, and then a user would be able to share an empty list (see issue 3).

### PR DESCRIPTION
**Pull request by lydiadunning**
## Description

This PR adds functionality allowing the user to connect to an existing list.

### User Facing Behavior
Adds a text input field to a form in the Home view, allowing a user to input a listToken. When this form is submitted, if the list exists the listToken is saved in local storage and the user is redirected to the list page. Otherwise, a message informs the user that the list was not found.

### Firebase API
Adds a function to firebase.js to return the count of items in a collection. If the collection does not exist, the count will be 0.

### Refactoring
Changed the AddItem view so that when a new item is added to a new list, firebase creates a new collection with the same name as the token in local storage.

## Related Issue

closes #5 

## Acceptance Criteria
If a user **doesn’t** already have a token:
- [x] The `Home` view shows a form that allows the user to enter a token to join an _existing_ list, in addition to the button that allows them to create a _new_ list.
- [x] The input that accepts the list token has a semantic `label` element associated with its
- [x] The user can submit this form with both the mouse and the `Enter` key
- [x] If the list exists,
	- the token is saved in `localStorage`
	- the user is redirected to the `List` view and shown the items on that list
- [x] If the list _does not_ exist, the user is shown an error message that explains the problem

If a user **does** already have a token:
- [x] They are automatically redirected to the `List` view.

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
| ✓   | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

### Before

image

### After

https://github.com/the-collab-lab/tcl-65-smart-shopping-list/assets/71429289/e0c7ae2c-d540-4c88-97c1-db332bc71ef8

## Testing Steps / QA Criteria
- [x] Open the preview link provided by GitHub actions.
- [x] Remove any listToken currently saved in browser local storage.
- [x] Enter 'flout woman waxy' in the input field to attempt to join an existing list 
- [x] Check that the new list is visible and you can add an item
- [x] Enter a different string in the input field to see that the list was not found.

### CODE REVIEW

**Review by patty-rose — APPROVED** [review]
PR looks great! Easy to understand testing steps, works well in GH preview, code is clean and logical. Great work team! You rock!!

**Review by shannonerhea — APPROVED** [review]
Working as expected. Awesome job!!!!

File: src/views/AddItem.jsx [hunk 1]
```diff
@@ -11,7 +11,7 @@ export function AddItem() {
 		const daysUntilNextPurchase = parseInt(anticipation);
 
 		try {
-			await addItem(`my test list`, {
+			await addItem(listToken, {
```
Comment (mjhcodes) [inline src/views/AddItem.jsx:14]: Nice bonus work! 🌟 I think I remember this feature as an optional "nice to have" in Issue #3. I double-checked Firebase and there it is! 🔥 This helps take the shopping list to the next level.

Screen Shot 2023-08-30 at 9 46 47 PM

File: src/views/Home.jsx [hunk 2]
```diff
@@ -1,7 +1,23 @@
 import './Home.css';
-import { Link } from 'react-router-dom';
+import { Link, Navigate } from 'react-router-dom';
+import { useState } from 'react';
+import { checkCount } from '../api/firebase';
+
+export function Home({ setListToken, handleNewToken }) {
+	const [inputToken, setInputToken] = useState('');
+	const [message, setMessage] = useState('');
```
Comment (mjhcodes) [inline src/views/Home.jsx]: It might help the reader's understanding if this were a bit more descriptive, such as `notFoundMessage`.

**Review by mjhcodes — APPROVED** [review]
Awesome job! 🙌 Hits all the ACs and then some.

**Pull request merged**